### PR TITLE
fix small mistake in transform documentation

### PR DESCRIPTION
**Pull request by alindgren**
Replace TransformValidationContext with TransformRouteValidationContext in TransformFactory example. I noticed it was wrong when I copied the code into a project I"m working on. It is correct in the sample at https://github.com/microsoft/reverse-proxy/blob/main/samples/ReverseProxy.Transforms.Sample/MyTransformFactory.cs

### CODE REVIEW

File: docs/docfx/articles/transforms.md [hunk 1]
```diff
@@ -1049,7 +1049,7 @@ services.AddReverseProxy()
 ```C#
 internal class MyTransformProvider : ITransformProvider
 {
-    public void Validate(TransformValidationContext context)
+    public void Validate(TransformRouteValidationContext context)
```
Comment (MihaZupan) [inline docs/docfx/articles/transforms.md:1052]:
```suggestion
    public void ValidateRoute(TransformRouteValidationContext context)
```

**Review by MihaZupan — COMMENTED** [review]
Thanks for bringing this up. Looking at the interface, we're also missing
```c#
void ValidateCluster(TransformClusterValidationContext context);
```
in this sample.

Can you please copy the working sample from
https://github.com/microsoft/reverse-proxy/blob/main/samples/ReverseProxy.Transforms.Sample/MyTransformProvider.cs to this doc?

**Comment by alindgren** [comment]
> @alindgren please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: docs/docfx/articles/transforms.md (hunk 1 above)
Comment (alindgren) [reply docs/docfx/articles/transforms.md]: @MihaZupan OK, I've copied the samples for both provider and transformer.

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**